packages/core/admin/ee/server/controllers/workflows/stages/index.js

```diff
@@ -121,4 +121,58 @@ module.exports = {
 
     ctx.body = { data: await sanitizeOutput(updatedEntity) };
   },
+
+  /**
+   * List all the stages that are available for a user to transition an entity to.
+   * If the user has permission to change the current stage of the entity every other stage in the workflow is returned
+   * @async
+   * @param {*} ctx
+   * @param {string} ctx.params.model_uid - The model UID of the entity.
+   * @param {string} ctx.params.id - The ID of the entity.
+   * @throws {ApplicationError} If review workflows is not activated on the specified model UID.
+   */
+  async listAvailableStages(ctx) {
+    const stagePermissions = getService('stage-permissions');
+    const workflowService = getService('workflows');
+
+    const { model_uid: modelUID, id } = ctx.params;
+
+    // Load entity
+    const entity = await strapi.entityService.findOne(modelUID, Number(id), {
+      populate: [ENTITY_STAGE_ATTRIBUTE],
+    });
+
+    if (!entity) {
+      ctx.throw(404, 'Entity not found');
+    }
+
+    const entityStageId = entity[ENTITY_STAGE_ATTRIBUTE]?.id;
+    const canTransition = stagePermissions.can(STAGE_TRANSITION_UID, entityStageId);
+
+    const workflowCount = await workflowService.count();
```

Comment:
This endpoint will block the Edit View first render, so I would try to do as many queries in parallel here! Could you gather all the queries that can be done in paralell inside a promise all?